include/tvm/node/container.h

```diff
@@ -1449,6 +1449,19 @@ namespace runtime {
 // Additional overloads for PackedFunc checking.
 template <typename T>
 struct ObjectTypeChecker<Array<T>> {
+  static Optional<String> CheckAndGetMismatch(const Object* ptr) {
+    if (ptr == nullptr) return NullOpt;
+    if (!ptr->IsInstance<ArrayNode>()) return String(ptr->GetTypeKey());
+    const ArrayNode* n = static_cast<const ArrayNode*>(ptr);
+    for (size_t i = 0; i < n->size(); i++) {
+      const ObjectRef& p = (*n)[i];
+      Optional<String> check_subtype = ObjectTypeChecker<T>::CheckAndGetMismatch(p.get());
+      if (check_subtype.defined()) {
+        return String("Array[index " + std::to_string(i) + ": " + check_subtype.value() + "]");
```

Comment:
Hey, I just wonder, if there is a variable `x` typed as nested `Array<Array<SomeType>>`, and some checking fails on like `x[0][1]`, in this case, what the error message could be?

Reply:
`Expected Array[Array[OtherType]] but got Array[index 0: Array[index 1: SomeType]]`

Reply:
Nice!